### ISSUE TITLE
With vim-fugitive, :Denite gitstatus in submodules causes an error

### ISSUE THREAD

**Issue by delphinus** [chemzqm/denite-git#17]
## tl;dr

With [vim-fugitive][], `:Denite gitstatus` causes an odd error. The reason is that `FugitiveExtractGitDir()` returns a different value from `denite#git#gitdir()`. To avoid this, denite-git should not use `b:git_dir` that comes from `FugitiveExtractGitDir()`.

[vim-fugitive]: https://github.com/tpope/vim-fugitive

## details

`denite#git#gitdir()` reuses pre-defined `b:git_dir`. The `b:git_dir` is calculated by `FugitiveExtractGitDir()`, you know.

But `FugitiveExtractGitDir()` has different behavior from `denite#git#gitdir()` about managing submodules. I have written an example for this.

```sh
~ $ git clone https://github.com/delphinus/test-denite-git
~ $ cd test-denite-git
# clone submodules placed on /sub/
~/test-denite-git $ git submodule update --init
# and open a file on the submodule
~/test-denite-git $ vim sub/README.md
```

And then, call the two funcs.

```vim
" previously delete the calculated value
:call remove(b:, 'git_dir')
:echo FugitiveExtractGitSir()
" => ~/test-denite-git/.git/modules/sub
:echo denite#git#gitdir()
" => ~/test-denite-git/.git
```

`:Denite gitstatus` expects that `denite#git#gitdir()` returns `~/test-denite-git/.git` and it runs `git status` on the parent dir: `~/test-denite-git`.

But in usual, Vim calls `FugitiveDetect()` on `BufReadPost` and sets `b:git_dir` which comes from `FugitiveExtractGitDir()`. After that, you call `:Denite gitstatus`, and `denite#git#gitdir()` returns `b:git_dir` . Then `git status` is called on the dir: `~/test-denite-git/.git/modules` and returns an error `fatal: this operation must be run in a work tree`.

At the last, an error below occurs (`KeyError: 'f'` means the first letter of `fatal: this...`).

<details><summary>error from <code>:Denite gitstatus</code></summary>

```
[denite] Traceback (most recent call last):
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/rplugin.py", line 25, in start
[denite]     return ui.start(args[0], context)
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/ui/default.py", line 81, in start
[denite]     self._start(context['sources_queue'][0], context)
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/ui/default.py", line 140, in _start
[denite]     self.gather_candidates()
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/ui/default.py", line 688, in gather_candidates
[denite]     self._denite.gather_candidates(self._context)
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/parent.py", line 20, in gather_candidates
[denite]     self._put('gather_candidates', [context])
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/parent.py", line 53, in _put
[denite]     return self._child.main(name, args, queue_id=None)
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/child.py", line 35, in main
[denite]     self.gather_candidates(args[0])
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/child.py", line 87, in gather_candidates
[denite]     source.context, source)
[denite]   File "/Users/jinnouchi.yasushi/test/denite.nvim/rplugin/python3/denite/child.py", line 324, in _gather_source_candidates
[denite]     candidates = source.gather_candidates(context)
[denite]   File "/Users/jinnouchi.yasushi/test/denite-git/rplugin/python3/denite/source/gitstatus.py", line 104, in gather_candidates
[denite]     candidates.append(_parse_line(line, root, winnr))
[denite]   File "/Users/jinnouchi.yasushi/test/denite-git/rplugin/python3/denite/source/gitstatus.py", line 30, in _parse_line
[denite]     index_symbol = STATUS_MAP[line[0]]
[denite] KeyError: 'f'
[denite] Please execute :messages command.
```
</details>

## conclusion

It can avoid this error by calculating the dir in `denite#git#gitdir()` whenever it is called, or it should cache into another variable. `b:git_dir` is a value for the different purpose. It should not use that.
